libraries/AP_Math/matrixN.cpp

```diff
@@ -48,7 +48,7 @@ void MatrixN<T,N>::force_symmetry(void)
 {
     for (uint8_t i = 0; i < N; i++) {
         for (uint8_t j = 0; j < (i - 1); j++) {
-            v[i][j] = (v[i][j] + v[j][i]) * 0.5;
+            v[i][j] = (v[i][j] + v[j][i]) * 0.5f;
```

Comment:
Technically shouldn't this actually be `/ 2` to respect the fact that it could be called with integer types? In practice I think we only have floating point matrices though, so maybe that's fine.

Godbolt indicates on any optimization level (O1-O3, and Os) that the compiler will correctly optimize the `/ 2` into a float multiplication if using a float, but if using integer types in the template it will stick with the `/ 2` which is faster then doing the conversions to/from float.

Reply:
Pushed up a patch to make this change.